src/dsp_tools/commands/xml_validate/prepare_input.py

```diff
@@ -0,0 +1,150 @@
+from pathlib import Path
+from typing import cast
+
+from lxml import etree
+
+from dsp_tools.commands.xml_validate.models.deserialised import LinkValueDeserialised
+from dsp_tools.commands.xml_validate.models.deserialised import ListValueDeserialised
+from dsp_tools.commands.xml_validate.models.deserialised import PermissionsDeserialised
+from dsp_tools.commands.xml_validate.models.deserialised import ProjectDeserialised
+from dsp_tools.commands.xml_validate.models.deserialised import ResourceDeserialised
+from dsp_tools.commands.xml_validate.models.deserialised import RichtextDeserialised
+from dsp_tools.commands.xml_validate.models.deserialised import SimpleTextDeserialised
+from dsp_tools.commands.xml_validate.models.deserialised import ValueDeserialised
+from dsp_tools.commands.xml_validate.models.xml_deserialised import PermissionsXML
+from dsp_tools.commands.xml_validate.models.xml_deserialised import ProjectXML
+from dsp_tools.commands.xml_validate.models.xml_deserialised import ResourceXML
+from dsp_tools.utils.xml_utils import parse_and_clean_xml_file
+from dsp_tools.utils.xml_validation import validate_xml
+
+
+def parse_file(file: Path) -> ProjectXML:
+    """Returns an object which follows the structure of the XML closely"""
+    root = _parse_file_validate_with_schema(file)
+    return _transform_into_xml_deserialised(root)
+
+
+def _parse_file_validate_with_schema(file: Path) -> etree._Element:
+    root = parse_and_clean_xml_file(file)
+    validate_xml(root)
+    return root
+
+
+def _transform_into_xml_deserialised(root: etree._Element) -> ProjectXML:
+    shortcode = root.attrib["shortcode"]
+    default_ontology = root.attrib["default-ontology"]
+    permissions = [_get_permissions(x) for x in root.iterdescendants(tag="permissions")]
+    resources = [_get_resources(x) for x in root.iterdescendants(tag="resource")]
+    return ProjectXML(
+        shortcode=shortcode, default_onto=default_ontology, permissions=permissions, xml_resources=resources
+    )
+
+
+def _get_resources(ele: etree._Element) -> ResourceXML:
+    values = list(ele.iterchildren())
+    ele_attribs = cast(dict[str, str], ele.attrib)
+    return ResourceXML(res_attrib=ele_attribs, values=values)
+
+
+def _get_permissions(permission_ele: etree._Element) -> PermissionsXML:
+    per_id = permission_ele.attrib["id"]
+    permissions = list(permission_ele.iterchildren())
+    return PermissionsXML(permission_id=per_id, permission_eles=permissions)
+
+
+def deserialise_xml_project(project: ProjectXML) -> ProjectDeserialised:
```

Comment:
this naming is misleading -`ProjectXML` is already deserialised. Also, I'm not yet really convinced that these two separate steps provide any benefits. But I'm happy to be convinced otherwise

Reply:
I understand why it seems redundant. I was thinking about doing this in one step too, but I did not want to do too much logic at the same time. What I could do is when I implement the two versions that I already do some changes here for example make the props into IRIs for SHACL etc. What do you think?

Reply:
I would prefer to have `ProjectDeserialized` be the first step in the pipeline and kick out the `ProjectXML` layer, so that we do not have any XML elements in code. Then it is consistent with the naming too, as it deserializes the input into an un-opinionated object. Any further step should then transform it into an object that indicates what was done (like IRI resolving etc.).